### PR TITLE
Configurable Job Reference Validation

### LINKED ISSUE #4527: Configurable Job Reference Validation
## Background
Job reference names and UUIDs can become out of sync; the reference name will not match the UUID name. For certain user use cases this may be undesirable and should result in validation errors.

This was originally introduced in #4366 , however it created a cascade of changes breaking backwards compatibility for project/job imports.

## Solution
We would like to re-introduce the validation as an option with the old, no validation, behavior as the default for backwards compatibility.

**API**
* Add validation option to job/project import actions
* Add UUID/Name stripping options to export actions

**GUI**
* Allow user to pick name or UUID as the key when creating job references
* If UUID is used we should look up and display the name; this should not look like a text input field
* Allow user to choose strict validation when importing projects/jobs
* Allow user to choose stripping reference UUIDs or names on export?

### PR DESCRIPTION
**Pull request by jtobard**
* Allow user to pick name or UUID as the key when creating job references:

imagen

* Allow user to choose strict validation when importing jobs:
screenshot at mar 05 12-10-21

* And importing projects:
screenshot at mar 05 12-11-01

* Optional param to API endpoint  `/api/V/project/[PROJECT]/jobs/import`: `validateJobref`

Fix #4527 

TODO:
Option to strip name or UUID on export, see #4285

### CODE REVIEW

File: rundeckapp/grails-app/views/execution/_wfItemView.gsp [hunk 1]
```diff
@@ -27,13 +27,13 @@
...
                 %{--Display job icon and name--}%
-                <g:set var="foundjob" value="${edit?null:ScheduledExecution.findScheduledExecution(item.jobGroup?item.jobGroup:null,item.jobName,item.jobProject?item.jobProject:project,item.uuid)}"/>
+                <g:set var="foundjob" value="${edit?null:(item.useName?ScheduledExecution.findScheduledExecution(item.jobGroup?item.jobGroup:null,item.jobName,item.jobProject?item.jobProject:project):ScheduledExecution.findScheduledExecution(null,null,null,item.uuid))}"/>
                 <g:if test="${foundjob}">
                 <g:link controller="scheduledExecution" action="show" id="${foundjob.extid}">
                     <g:if test="${!noimgs }">
                         <i class="glyphicon glyphicon-book"></i>
                     </g:if>
-                    <g:enc>${(item.jobGroup?item.jobGroup+'/':'')+(item.jobName?:item.uuid) + (item.jobProject?' (' + item.jobProject+') ':'') }</g:enc></g:link>
+                    <g:enc>${foundjob.generateFullName()}${(item.jobProject?' (' + item.jobProject+') ':'') }</g:enc></g:link>
```
Comment (gschueler) [inline rundeckapp/grails-app/views/execution/_wfItemView.gsp]: `foundjob` might be null?

File: rundeckapp/grails-app/views/scheduledExecution/_uploadForm.gsp [hunk 2]
```diff
@@ -189,6 +189,19 @@
                 <div class="help-block"><g:message code="project.archive.import.jobUuidOption.remove.description"/></div>
               </div>
             </div>
+            <div class="form-group">
+              <div class="control-label text-form-label">Validate Referenced Jobs:</div>
+              <div class="radio">
+                <input type="radio" name="validateJobref" value="false" checked/>
+                <label title="Original UUIDs will be preserved, conflicting UUIDs will be replaced"><g:message code="archive.import.importJobRef.false.title"/></label>
```
Comment (gschueler) [inline rundeckapp/grails-app/views/scheduledExecution/_uploadForm.gsp]: seems the wrong `title=` content

File: rundeckapp/grails-app/views/scheduledExecution/_uploadForm.gsp [hunk 3]
```diff
@@ -189,6 +189,19 @@
...
+                <div class="help-block"><g:message code="archive.import.importJobRef.false.help"/></div>
+              </div>
+              <div class="radio">
+                <input type="radio" name="validateJobref" value="true"/>
+                <label title="New UUIDs will be generated for every imported Job"><g:message code="archive.import.importJobRef.true.title"/></label>
```
Comment (gschueler) [inline rundeckapp/grails-app/views/scheduledExecution/_uploadForm.gsp]: seems the wrong `title=` content

File: rundeckapp/grails-app/views/execution/_wfItemView.gsp (hunk 1 above)
Comment (jtobard) [reply rundeckapp/grails-app/views/execution/_wfItemView.gsp]: this field can be null if the referenced job does not exist, it has been deleted or renamed. But it is already controlled in line 31

**Pull request merged**